build-scripts/src/index.ts

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env node
+import fs, { promises as fsp } from 'fs';
+import { program } from 'commander';
+import queryGitHub from './github';
+import querySecurityScorecard from './security-scorecard';
+
+program
+  .version('1.0.0', '-v, --version')
+  .requiredOption('-p, --path <string>', 'The path where the files should be saved.')
+  .requiredOption('-o, --org <string>', 'The github organisation.')
+  .requiredOption('--gh_priv_key <string>', 'The github private key for the app.')
+  .requiredOption('--gh_app_id <string>', 'The github id of the app.')
+  .requiredOption('--gh_app_install_id <string>', 'The github id for the installed app.')
+  .parse();
+const options = program.opts();
+
+type RepoId = string;
+type DateIsoString = string;
+interface IndexData {
+  id: RepoId;
+  name: string;
+  rating: number | null;
+  createdAt: DateIsoString;
+  updatedAt: DateIsoString;
+}
+(async (): Promise<void> => {
+  await ensureFolderExists(options.path)
+
+  const index: IndexData[] = [] // main doc, used for searches & pagination
+  const queryGitHubArgs = {
+    org: options.org,
+    itemsPerPage: 10,
+    gh_priv_key: options.gh_priv_key,
+    gh_app_id: options.gh_app_id,
+    gh_app_install_id: options.gh_app_install_id
+  }
+
+  for await (const gitHubItem of queryGitHub(queryGitHubArgs)) {
```

Comment:
This is a simple thought.
I don't know how many repositories we are and if there is a rate limit in these APIs, but using an asyncIterator, we call the API every iteration. I think this affects the execution time and the number of minutes for the GitHub Actions increases. Understanding if it's possible to resolve this issue using a `Promise.all` or a `Promise.allSettled` to "parallelise" all the requests and then manipulate the data after.
I don't want to refactor the codebase now; I just want to raise the flag on this point. This is impossible if we have a rate limit in the API, but maybe understanding this solution could be helpful. We can also go ahead with the current solution, check the time and the performance, and refactor the codebase in the future.